tests/test_array.py

```diff
@@ -870,6 +870,30 @@ def test_write_empty_chunks_behavior(
         assert arr.nchunks_initialized == arr.nchunks
 
 
+@pytest.mark.parametrize("store", ["memory"], indirect=True)
+@pytest.mark.parametrize("fill_value", [0.0, -0.0])
+@pytest.mark.parametrize("dtype", ["f4", "f2"])
+def test_write_empty_chunks_negative_zero(
+    zarr_format: ZarrFormat, store: MemoryStore, fill_value: float, dtype: str
+) -> None:
+    # regression test for https://github.com/zarr-developers/zarr-python/issues/3144
+
+    arr = zarr.create_array(
+        store=store,
+        shape=(2,),
+        zarr_format=zarr_format,
+        dtype=dtype,
+        fill_value=fill_value,
+        chunks=(1,),
+        config={"write_empty_chunks": False},
+    )
+    assert arr.nchunks_initialized == 0
+
+    # initialize the with the negated fill value (-0.0 for +0.0, +0.0 for -0.0)
+    arr[:] = -fill_value
+    assert arr.nchunks_initialized == arr.nchunks
```

Comment:
this test is fine but ideally we would be testing the altered function explicitly, instead of indirectly via array creation + chunk writing. this is not a blocker for this PR, just something to sort out down the road

Reply:
That test is basically copied from `test_write_empty_chunks_behavior` right above it. But yeah, it might be worth to have both a unit and an integration test in this case (: